### PR TITLE
add more c++ and R tests

### PR DESCRIPTION
**Pull request by Bai-Li-NOAA**
Thanks for opening a pull request (PR) to FIMS! Please describe the changes under the headings below. Please also go through the checklist below to ensure the appropriate steps have been taken during development. For smaller changes, feel free to skip steps that aren't relevant, but at least describe the feature under the "what is the feature".

# What is the feature?
Add more C++ and R unit tests for FIMS.

# How have you implemented the solution?
Add tests for files listed in the issue #182. However, I have not been able to figure out the reason why codecov ignores tests/gtest/test_sr_beverton_holt.cpp.

Regarding R/io.R file, it might be good to create a yml file to run mkObj() instead of writing an R unit test. It takes 6 hours to run mkObj() through the r-cmd-check (relevant issue #247). 

# Does it impact any other area of the project?
No

# Does this change impact the model input or output?*
No

# How to test this change
All GHA checks passed: https://github.com/NOAA-FIMS/FIMS/actions?query=branch%3Atest-add-more-tests

Codecov only shows updated coverage for C++ tests. The coverage from inst/ increases from 37.28% to [93.90%](https://app.codecov.io/gh/NOAA-FIMS/FIMS/tree/test-add-more-tests).

# Developer pre-PR Checklist

Please do these steps locally for big changes. Note GitHub Actions does all of these checks automatically when pushing to the repository. Please see [code development section in the contributor guide](https://noaa-fims.github.io/collaborative_workflow/contributor-guidelines.html#code-development) for details.

- [x] Run [cmake build and ctest locally](https://noaa-fims.github.io/collaborative_workflow/testing.html#c-unit-testing-and-benchmarking) and make sure the C++ tests pass
- [ ] Run `devtools::document()` locally and push changes to the remote feature branch
- [ ] Run `styler::style_pkg()` locally and push changes to remote feature branch. If there are many changes, please do this in a separate commit.
- [x] Run `devtools::check()` locally and make sure the package can be compiled and R tests pass. If there are failing tests, run `devtools::test(filter = "file_name")` (where "test-file_name.R" is the testthat file containing failing tests) and edit code/tests to troubleshoot tests.
- [x] Before opening the PR, make sure all the github actions are passing on the remote feature branch.
- [x] Make sure this PR has an informative title.

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
Thanks, @Bai-Li-NOAA, this is on my to do list. Please feel free to ping me about it, though, if it doesn't get done within the next few days!

File: tests/gtest/test_fims_math_ad_fabs_min_max.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace
+{
+
+  // Test ad_fabs
+  TEST(ad_fabs, use_double_values)
+  {
+    // Expected value from R: x=2; sqrt(x*x+1e-5) = 2.000002
+    EXPECT_NEAR(fims::ad_fabs(2.0), 2.000002, 0.000001);
+    EXPECT_NE(fims::ad_fabs(2.0), 1.0);
```
Comment (k-doering-NOAA) [inline tests/gtest/test_fims_math_ad_fabs_min_max.cpp:12]: should this be `EXPECT_NEAR` or is `EXPECT_NE` correct?

Comment (k-doering-NOAA) [reply tests/gtest/test_fims_math_ad_fabs_min_max.cpp:12]: I just read some documentation and realized that `EXPECT_NE` means expect not equal

**Review by k-doering-NOAA — APPROVED** [review]
@Bai-Li-NOAA , looks good! I ran the testthat and GoogleTest tests locally, and all passed. Apologize for the delay on this review.

I made one comment for you to check, but I'm pretty sure you can dismiss it. It would be great to merge this in.

I see there were a few comments about issues that weren't resolved with this PR - I think those should just be noted in the issue or a separate issue can be opened for them (I didn't check to see if they were captured).

File: tests/gtest/test_fims_math_ad_fabs_min_max.cpp (hunk 1 above)
Comment (Bai-Li-NOAA) [reply tests/gtest/test_fims_math_ad_fabs_min_max.cpp:12]: You are correct!

**Pull request merged**